src/map/ai/helpers/pathfind.cpp

```diff
@@ -434,12 +435,9 @@ void CPathFind::StepTo(const position_t& pos, bool run)
         }
     }
 
-    m_POwner->loc.p.moving += (uint16)((0x36 * ((float)m_POwner->speed / 0x28)) - (0x14 * (mode - 1)));
+    m_POwner->loc.p.moving += speedChange ? 0x28 : 0x35;
 
-    if (m_POwner->loc.p.moving > 0x2fff)
-    {
-        m_POwner->loc.p.moving = 0;
-    }
+    m_POwner->loc.p.moving %= 0x2001;
```

Comment:
This is 13 bits, but I saw it going from 8179 to 39. This could be `(8179 + 53) % 8193` or `(8179 + 52) % 8192`. I went with the former to stick with the 53, but I guess the latter makes more sense if you consider the +/-1 variance. It seems to work fine though so I don't think it matters either way (this way just double counts 0).